src/reconcile/rec_write.c

```diff
@@ -218,8 +218,11 @@ __reconcile(WT_SESSION_IMPL *session, WT_REF *ref, WT_SALVAGE_COOKIE *salvage, u
     /* Wrap up the page reconciliation. */
     if (ret == 0 && (ret = __rec_write_wrapup(session, r, page)) == 0)
         __rec_write_page_status(session, r);
-    else
+    else {
+        /* Make sure that reconciliation doesn't free the page that has been written to disk. */
+        WT_ASSERT(session, page->dsk == NULL || ref->addr != NULL);
```

Comment:
@michaelcahill, @keithbostic I added the above assert to ensure that the in-memory page has the disk address and ref doesn't. It is possible to clear the ref->addr in the page delete scenario, but that reconciliation shouldn't fail.

Reply:
The assert fired in our PR testing, Will look into it and update.

Reply:
I changed the assert as if there exists an address and after wrapup, it should have the address also in the error scenario.